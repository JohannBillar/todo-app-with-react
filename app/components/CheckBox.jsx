import React, { PropTypes } from 'react';

const propTypes = {
  isChecked: PropTypes.bool.isRequired,
  toggleCheckbox: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  children: PropTypes.string.isRequired,
};

function CheckBox({ isChecked, toggleCheckbox, id, children }) {
  return (
    <label htmlFor={id}>
      <input
        className="checkbox-input"
        type="checkbox"
        checked={isChecked}
        onChange={toggleCheckbox}
        id={id}
      />
      {children}
    </label>
  );
}

CheckBox.propTypes = propTypes;

export default CheckBox;
