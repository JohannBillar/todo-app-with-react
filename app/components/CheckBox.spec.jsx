import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CheckBox from './CheckBox';
import renderer from 'react-test-renderer';
 
test('CheckBox should render a checkbox', () => {
  const props = {
    isChecked: false,
    id: 0,
    children: 'test.number',
    toggleCheckbox: jest.fn(),
  };

  // let toggleCheckbox = jest.fn();
  // toggleCheckbox.mockReturnValueOnce(true);

  const component = shallow(
    <CheckBox
      {...props}
    />
  );

  let tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
