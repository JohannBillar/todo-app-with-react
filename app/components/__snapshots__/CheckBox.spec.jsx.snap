exports[`test CheckBox should render a checkbox 1`] = `
<label
  htmlFor={0}>
  <input
    checked={false}
    className="checkbox-input"
    id={0}
    onChange={[Function]}
    type="checkbox" />
  test.number
</label>
`;
